torch/_functorch/aot_autograd.py

```diff
@@ -2106,13 +2108,13 @@ def runtime_wrapper(*args):
                 all_outs = call_func_with_args(
                     compiled_fn,
                     args,
-                    disable_amp=True,
```

Comment:
cc @eellison do you remember why we set disable_amp=True at runtime to begin with?

This PR changes it to instead only disable amp if we saw that amp was enabled during tracing. At first I thought that would be wrong, but it seems like dynamo properly guards on amp being set.

Reply:
I don't. In the original pr where I added it we only disabled if amp was enabled: https://github.com/pytorch/pytorch/blame/master/torch/_functorch/aot_autograd.py#L1287-L1288. I'm not familiar with these new lines.

Reply:
It is found that `disable_amp=True` was introduced in #92857.

Reply:
oh you're right. Thanks, this seems ok!